### PR TITLE
Improve validation for some fields with enumerations

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Adds validation for a few properties that have a fixed set of values, ensuring errors are reported by kubectl instead of waiting for an ARM failure.

Followup to #1483 and #1485

**How does this PR make you feel**:
[#spiffy](https://media.giphy.com/media/gg2sGd9VH2QQg2Lq4D/giphy.gif)

### CODE REVIEW

File: api/v1alpha2/sharedazuredbs_types.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ const (
 	PSQLMemoryOptimized SkuTier = "MemoryOptimized"
 )
 
+// +kubebuilder:validation:Enum=Enabled;Disabled
```
Comment (matthchr) [inline api/v1alpha2/sharedazuredbs_types.go:54]: minor: You have two formats here, the shorter/simpler one and the longer/more verbose one. Feels like we should have a single format we prefer?

In most of the existing ASO code it seems to be the `Enabled;Disabled` format, but the generator is doing `{"Enabled","Disabled"}`. Unsure which we should pick but feels like 1 style per file is better?

**Review by matthchr — APPROVED** [review]

**Review by babbageclunk — APPROVED** [review]

File: api/v1alpha2/sharedazuredbs_types.go (hunk 1 above)
Comment (theunrepentantgeek) [reply api/v1alpha2/sharedazuredbs_types.go:54]: This syntax works, and I didn't get the other syntax to work in my testing - so I'd prefer to leave this as is.

Comment (theunrepentantgeek) [reply api/v1alpha2/sharedazuredbs_types.go:54]: (Which is not to say the more verbose syntax is wrong, maybe I had a local issue.)

Comment (matthchr) [reply api/v1alpha2/sharedazuredbs_types.go:54]: What about the more verbose syntax you use above?
// +kubebuilder:validation:Enum={"Basic","GeneralPurpose","MemoryOptimized"}

**Pull request merged**